### PR TITLE
#15363. Add escaping for JSON double-quoted strings

### PR DESCRIPTION
**Pull request by sergiohs84**
...and unescape them for `usabd` command

### CODE REVIEW

File: src/command.cpp [hunk 1]
```diff
@@ -77,11 +77,26 @@ void Command::arg(const char* name, const char* value, int quotes)
...
     if (quotes)
     {
-        json.append("\"");
+        json.append("\":\"");
+        size_t valueLen = strlen(value);
+        for (size_t i = 0; i < valueLen; i++)
+        {
+            if (value[i] == '\"'
+                    || value[i] == '\\'
+                    || value[i] == '/')
```
Comment (mattw-mega) [inline src/command.cpp:88]: According to the JSON spec (https://www.ietf.org/rfc/rfc4627.txt, Strings section), we don't need to escape /, but we do need to escape any control codes in the string.

File: src/command.cpp [hunk 2]
```diff
@@ -77,11 +77,26 @@ void Command::arg(const char* name, const char* value, int quotes)
...
     json.append("\"");
     json.append(name);
-    json.append(quotes ? "\":\"" : "\":");
-    json.append(value);
     if (quotes)
     {
-        json.append("\"");
+        json.append("\":\"");
+        size_t valueLen = strlen(value);
+        for (size_t i = 0; i < valueLen; i++)
```
Comment (mattw-mega) [inline src/command.cpp:84]: Do we need to take UTF-8 into account here, and increment by character size in bytes?  eg might we alter a character that takes two or more bytes to represent but isn't one of those below?

File: src/command.cpp (hunk 1 above)
Comment (sergiohs84) [reply src/command.cpp:88]: Yep, according to the spec those are the chars that should be escaped for Strings:
```
RFC 4627                          JSON                         July 2006

         string = quotation-mark *char quotation-mark

         char = unescaped /
                escape (
                    %x22 /          ; "    quotation mark  U+0022
                    %x5C /          ; \    reverse solidus U+005C
                    %x2F /          ; /    solidus         U+002F
                    %x62 /          ; b    backspace       U+0008
                    %x66 /          ; f    form feed       U+000C
                    %x6E /          ; n    line feed       U+000A
                    %x72 /          ; r    carriage return U+000D
                    %x74 /          ; t    tab             U+0009
                    %x75 4HEXDIG )  ; uXXXX                U+XXXX
```
I only added the printable chars with the specific case we aim to solve in mind. But a proper escaping should include all the above, indeed.

File: src/command.cpp (hunk 2 above)
Comment (sergiohs84) [reply src/command.cpp:84]: This is a good point. The specs say:
```The representation of strings is similar to conventions used in the C family of programming languages. A string begins and ends with quotation marks. All Unicode characters may be placed within the quotation marks except for the characters that must be escaped: quotation mark, reverse solidus, and the control characters (U+0000 through U+001F).```
Hence, it seems like UTF-8 chars don't need (and should not) be escaped, unless they are the `"`, the `\` or control characters.

**Comment by sergiohs84** [comment]
Since it's been agreed the JSON parser in the SDK should not be changed, but API to enforce B64 for any string, this PR is closed.

**Pull request closed without merging**